projects/dsp-ui/src/lib/action/directives/external-links/external-links.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Directive, HostBinding, Inject, Input, OnChanges, PLATFORM_ID } from '@angular/core';
+
+@Directive({
+  selector: 'a[href]'
+})
+export class ExternalLinksDirective implements OnChanges {
+
+    @HostBinding('attr.rel') relAttr = '';
+    @HostBinding('attr.target') targetAttr = '';
+    @HostBinding('attr.href') hrefAttr = '';
+    @HostBinding('style.color') fontColor = 'primary';
+    @Input() href: string;
+
+    // to check if we are running on the server, give a token value
+    constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+    ngOnChanges() {
+        this.hrefAttr = this.href;
+
+        if (this.isLinkExternal()) {
+            // makes sure that the new browser tab does not run on the same process and prevent it from accessing window.opener
+            this.relAttr = 'noopener';
+            // open the page in a new tab
+            this.targetAttr = '_blank';
+        }
+    }
+
+    /**
+     * Check if the link opens an external page
+     */
+    private isLinkExternal() {
```

Comment:
What do you mean? here I deal with external links only (= to another website/external page).

Reply:
I mean that a link to "http://rdfh.ch/0001/a-thing" would be detected as an external link although in fact it is internal. However, this will only be an issue if such links to resource Iris occur outside of the context of a text value with markup or a link value (for these two cases, we have custom components).

Reply:
if Knora resource Iris are not put directly as links in the template, e..g, `<a href="http://rdfh.ch/0001/a-thing">Resource</a>`, I think this won't be an issue.